lua/astronvim/utils/ui.lua

```diff
@@ -81,12 +81,15 @@ end
 
 --- Toggle buffer semantic token highlighting for all language servers that support it
 ---@param bufnr? number the buffer to toggle the clients on
-function M.toggle_buffer_semantic_tokens(bufnr)
+--- @param silent? boolean if true then don't sent a notification
+function M.toggle_buffer_semantic_tokens(bufnr, silent)
```

Comment:
I like the idea of adding `silent`, but we should do this for all of the functions at once and not start misaligning these function definitions

Reply:
Is that really necessary given that there's already a global check for `ui_notifications_enabled`? I only added it for `toggle_buffer_semantic_tokens` since it made sense for that specific situation.

Reply:
It's good for these functions to have silent options if people want to script with this API for whatever reason. Seems to be a good option in this API